src/Nncase.Core/IR/Expr.cs

```diff
@@ -86,6 +86,11 @@ public IRType CheckedType
         _ => throw new InvalidOperationException("Expr don't have a valid tensor type"),
     };
 
+    /// <summary>
+    /// Gets or sets outputs names.
+    /// </summary>
+    public List<string> OutputsNames { get; set; } = new();
```

Comment:
- Add a mutable `IRMetadata` class to maintain all metadata including outputs names.
- Use `IReadOnlyList` instead of `List`, output names should be set at once.